### PR TITLE
@craigspaeth => [Venice] Banner for venice articles

### PR DESCRIPTION
**Pull request by eessex**
- Closes https://github.com/artsy/force/issues/1328, Adds banner for 2017 content on 2015 Venice articles and section
- Also fixes styling for Follow buttons when not logged in 

screen shot 2017-05-10 at 11 07 21 am

screen shot 2017-05-10 at 11 07 10 am

### CODE REVIEW

File: desktop/apps/articles/templates/section.jade [hunk 1]
```diff
@@ -23,3 +23,6 @@ block body
         hr
         +share(section.get('title') + ' on Artsy')
   include full_feed
+
+  if veniceBanner != undefined
```
Comment (craigspaeth) [inline desktop/apps/articles/templates/section.jade]: Pro tip: You can rely on ["falseyness"](http://stackoverflow.com/questions/19839952/all-falsey-values-in-javascript) here and just use `if veniceBanner`.

**Comment by craigspaeth** [comment]
LGTM! Clean, clear with tests 👍  Feel free to merge when 💚

File: desktop/apps/articles/templates/section.jade (hunk 1 above)
Comment (eessex) [inline desktop/apps/articles/templates/section.jade]: Recently was told it is faster to do it this way, do you think there is a difference?

**Comment by eessex** [comment]
Just realized I forgot a method to close the pop-up -- going to add that now ¯\_(ツ)_/¯

File: desktop/apps/articles/templates/section.jade (hunk 1 above)
Comment (craigspaeth) [inline desktop/apps/articles/templates/section.jade]: Short answer:
The difference is so negligible that I would always pick the more readable/easier-to-understand solution.

Long answer:
It makes sense that Javascript is more performant at checking if something isn't equal to `undefined` vs. checking a truthy/falsey value. That said, _premature optimization is the root of all evil_ as [they say](https://shreevatsa.wordpress.com/2008/05/16/premature-optimization-is-the-root-of-all-evil/). The speed of JS execution is almost never a performance bottleneck unless you're doing something really intense like [parsing and compiling large graphs of stylus files on the fly](https://github.com/artsy/force/pull/1246). 95% of the performance battle with JS web apps has nothing to do with the execution speed of the language and is more about remembering not to use [synchronous APIs](https://nodejs.org/api/fs.html#fs_fs_readfilesync_file_options) within request handlers, [streaming](https://nodejs.org/api/stream.html) large data, optimizing data modeling and indexes, and keeping asset [bundle sizes small](https://github.com/artsy/force/pull/1402).

File: desktop/apps/article/test/client/index.coffee [hunk 2]
```diff
@@ -149,3 +152,8 @@ describe 'ArticleIndexView', ->
       $('.articles-promoted__name').text().should.equal 'Whitney Art Party'
       $('.articles-promoted__explore-button').text().should.equal 'Explore Auction'
       $('.articles-promoted__explore').attr('href').should.equal '/sale/whtney-art-party'
+
+  describe 'Venice section articles', ->
+    it '#closeVeniceBanner closes the banner', ->
+      $('.venice-redirect-banner a.icon-close').click()
+      console.log $.fn.fadeOut.called.should.eql true
```
Comment (craigspaeth) [inline desktop/apps/article/test/client/index.coffee]: Looks like a stray console.log here

File: desktop/apps/articles/templates/section.jade (hunk 1 above)
Comment (eessex) [inline desktop/apps/articles/templates/section.jade]: This makes a lot of sense, thanks for these resources!

File: desktop/apps/article/stylesheets/index.styl [hunk 3]
```diff
@@ -2,6 +2,7 @@
 @require '../../../components/articles_grid/stylesheets'
 @require '../../../components/tooltips'
 @require '../../../components/channel_nav'
+@require '../../articles/stylesheets/venice_banner'
```
Comment (craigspaeth) [inline desktop/apps/article/stylesheets/index.styl]: Noticing the repeated `closeVeniceBanner` methods in this PR and the requiring backward into another app, I think this is indicating `venice_banner` should be moved to the ./desktop/components folder where you can co-locate the stylus, jade template, and a tiny Backbone view that basically looks [like this view](https://github.com/artsy/force/pull/1403/files#diff-803cd0894b2ea6f609c14ab2c45e4d6b) you wrote. That way you can just require the component across apps and not have to repeat yourself. Let me know if that doesn't make sense and happy to explain more IRL.

Comment (eessex) [inline desktop/apps/article/stylesheets/index.styl]: Yeah makes alot of sense -- have now refactored this as a separate component

**Comment by craigspaeth** [comment]
Thanks for the refactor! LGTM now, feel free to merge when 💚

**Pull request merged**